src/SuperDumpService/Helpers/SuperDumpLogExtension.cs

```diff
@@ -0,0 +1,92 @@
+				bundleId, GetCustomPropertyString(customProperties).ToString(), uri);
+		}
+
+		public static void LogSearch(this ILogger logger, string text, HttpContext context, string filter) {
+			logger.LogInformation(SearchLogText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name, filter);
+		}
+
+		public static void LogNotFound(this ILogger logger, string text, HttpContext context, string identifierDescription, string identifier) {
+			logger.LogInformation(NotFoundText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name,
+				identifierDescription, identifier);
+		}
+
+		public static void LogElasticClean(this ILogger logger, string text, HttpContext context, bool clean) {
+			logger.LogInformation(ElasticCleanText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name, clean);
+		}
+		public static void LogLoginEvent(this ILogger logger, string text, HttpContext context, string username) {
+			logger.LogInformation(DefaultLogText, text, context.Connection.RemoteIpAddress.ToString(), username);
+		}
+
+		public static void LogFailedLogin(this ILogger logger, string text, HttpContext context, string username) {
+			logger.LogWarning(DefaultLogText, text, context.Connection.RemoteIpAddress.ToString(), username);
+		}
+
+		public static void LogDefault(this ILogger logger, string text, HttpContext context) {
+			logger.LogInformation(DefaultLogText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name);
+		}
+
+		public static void LogAccessDenied(this ILogger logger, string text, HttpContext context, string url) {
+			logger.LogWarning(AccessDeniedText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name, url);
+		}
+
+		public static void LogRequest(this ILogger logger, HttpContext context) {
+			logger.LogInformation(RequestLogText,
+				context.Connection.RemoteIpAddress.ToString(),
+				context.User.Identity != null ? context.User.Identity.Name : "null",
+				context.Request.Method, context.Request.Path, context.Request.Query);
+		}
+
+		public static void LogSimilarityEvent(this ILogger logger, string text, HttpContext context) {
+			logger.LogInformation(DefaultLogText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name);
+		}
+		public static void LogSimilarityEvent(this ILogger logger, string text, HttpContext context, string bundleId1, string dumpId1, string bundleId2, string dumpId2) {
+			logger.LogInformation(DumpComparisonText, text, context.Connection.RemoteIpAddress.ToString(), context.User.Identity.Name, bundleId1, dumpId1, bundleId2, dumpId2);
+		}
+
+		private static string GetCustomPropertyString(Dictionary<string, string> customProperties) {
+			StringBuilder sb = new StringBuilder();
+			foreach (var customProperty in customProperties) {
+				sb.AppendFormat("{0}: {1}, ", customProperty.Key, customProperty.Value);
+			}
+			sb.Remove(sb.Length - 2, 2);
```

Comment:
doing it like this feels a bit unclean.

using string.Join would be more elegant IMHO.

e.g.:
  return string.Join(", ", customProperties.Select(entry => $"{entry.Key}: {entry.Value}"));

I have not tested the statement above and it might be a bit less efficient (or is it?), but feels like cleaner code. So, if we're not using this in a tight loop with large dictionaries, I'd prefer the cleaner version.